Emby.Server.Implementations/Library/Resolvers/PhotoResolver.cs

```diff
@@ -68,10 +66,10 @@ protected override Photo Resolve(ItemResolveArgs args)
                 {
                     if (IsImageFile(args.Path, _imageProcessor))
                     {
-                        var filename = Path.GetFileNameWithoutExtension(args.Path);
+                        var filename = Path.GetFileNameWithoutExtension(args.Path.AsSpan());
 
                         // Make sure the image doesn't belong to a video file
-                        var files = _directoryService.GetFiles(Path.GetDirectoryName(args.Path));
+                        var files = _directoryService.GetFiles(Path.GetDirectoryName(args.Path) ?? throw new InvalidOperationException("Path can't be a root directory."));
```

Comment:
IMO `x ?? throw` should be on it's own line, but maybe this check is unnecessary anyways, since we check `args.IsDirectory` earlier

Reply:
If the directory name is null, there must be something very wrong with the path as GetDirectoryName gets you everything until the first `/`  or `\`. In that case we should return null and log the issue as debug.